crates/binjs_shared/src/json_conversion.rs

```diff
@@ -26,30 +28,18 @@ impl FromJSON for bool {
 impl FromJSON for f64 {
     fn import(value: &JSON) -> Result<Self, FromJSONError> {
         // JSON.as_f64() doesn't keep the precision.
-        match value {
-            JSON::Number(n) => {
-                if n.is_nan() {
-                    Ok(std::f64::NAN)
-                } else {
-                    let (positive, mantissa, exponent) = n.as_parts();
-                    let as_str = if positive {
-                        format!("{}e{}", mantissa, exponent)
-                    } else {
-                        format!("-{}e{}", mantissa, exponent)
-                    };
-                    Ok(as_str.parse::<f64>().expect("cannot parse the number!"))
-                }
-            }
-            _ => Err(FromJSONError {
-                expected: "Number".to_string(),
-                got: value.to_string(),
-            }),
+        if value.is_null() {
+            return Ok(std::f64::NAN);
         }
+        value.as_f64().ok_or_else(|| FromJSONError {
```

Comment:
nice that the hack for manual conversion gets removed :)
can you remove `JSON.as_f64() doesn't keep the precision.` comment above?

Reply:
Thanks, done.